### DISCUSSION TITLE
ReturnIf feature to reduce nesting

### DISCUSSION THREAD

**Discussion by ahmedameen2** [dotnet/csharplang#8509] [Language Ideas]
If you added language features like ReturnIfTrue, ReturnIfFalse, ReturnIfNull the code will be more readable and nesting will be reduced.

**Comment by HaloFour** [comment]
See: #2195

The language already allows you to put such a statement onto a single line with no nesting:

```cs
if (condition) return;
```

**Reply by ahmedameen2** [reply]
I was hoping to reduce the below in one line:
 var locationResult = await ValidateLocation(locationId);
            if (locationResult.IsFailed)
                return locationResult;

**Reply by HaloFour** [reply]
You could do this:

```cs
if ((await ValidateLocation(locationId)) is var locationResult && locationResult.IsFailed) return locationResult;
```

Although I question the readability of trying to jam that into a single line even with a bespoke language feature to do it.

It also seems like you're trying to work with a variant of a `Result<T>` ADT, and that's something that might be considered as a part of [discriminated unions](https://github.com/dotnet/csharplang/issues/113).

Not to mention wrapping a `Result<T>` in a `Task<T>` seems redundant.
